polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -2101,6 +2098,69 @@ MLIRScanner::EmitGPUCallExpr(clang::CallExpr *expr) {
   return std::make_pair(ValueCategory(), false);
 }
 
+static NamedAttrList getSYCLMethodOpAttrs(OpBuilder &builder,
+                                          mlir::Type funcType,
+                                          llvm::StringRef funcTypeName,
+                                          llvm::StringRef name,
+                                          llvm::StringRef mangledName) {
+  NamedAttrList attrs;
+  attrs.set(mlir::sycl::SYCLDialect::getBaseTypeAttrName(),
+            mlir::TypeAttr::get(funcType));
+  attrs.set(mlir::sycl::SYCLDialect::getBaseTypeNameAttrName(),
+            FlatSymbolRefAttr::get(builder.getStringAttr(funcTypeName)));
+  attrs.set(mlir::sycl::SYCLDialect::getFunctionNameAttrName(),
+            FlatSymbolRefAttr::get(builder.getStringAttr(name)));
+  attrs.set(mlir::sycl::SYCLDialect::getMangledFunctionNameAttrName(),
+            FlatSymbolRefAttr::get(builder.getStringAttr(mangledName)));
+  return attrs;
+}
+
+llvm::Optional<SYCLMethodOpInterface>
+MLIRScanner::createSYCLMethodOp(llvm::StringRef name, mlir::ValueRange operands,
```

Comment:
`name` --> `methodName` ?  Because the name provided is expected to be the name of a particular member function in a SYCL type.
Similarly `mangledName` -> `mangledMethodName` ?

Reply:
That sounds better. So many names...